src/altinn-app-frontend/src/components/base/ButtonComponent/ButtonComponent.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+import { useAppDispatch, useAppSelector } from 'src/common/hooks';
+import css from 'src/components/base/ButtonComponent/ButtonComponent.module.css';
+import { getComponentFromMode } from 'src/components/base/ButtonComponent/getComponentFromMode';
+import { SaveButton } from 'src/components/base/ButtonComponent/SaveButton';
+import { SubmitButton } from 'src/components/base/ButtonComponent/SubmitButton';
+import { FormDataActions } from 'src/features/form/data/formDataSlice';
+import type { IComponentProps } from 'src/components';
+import type { ButtonMode } from 'src/components/base/ButtonComponent/getComponentFromMode';
+import type { ILayoutCompButton } from 'src/features/form/layout';
+import type { IAltinnWindow } from 'src/types';
+
+export interface IButtonProvidedProps
+  extends IComponentProps,
+    ILayoutCompButton {
+  disabled: boolean;
+  id: string;
+  mode?: ButtonMode;
+  taskId?: string;
+}
+
+export const ButtonComponent = ({ mode, ...props }: IButtonProvidedProps) => {
+  const dispatch = useAppDispatch();
+  const autoSave = useAppSelector(
+    (state) => state.formLayout.uiConfig.autoSave,
+  );
+  const submittingId = useAppSelector((state) => state.formData.submittingId);
+  const savingId = useAppSelector((state) => state.formData.savingId);
+  const ignoreWarnings = useAppSelector(
+    (state) => state.formData.ignoreWarnings,
+  );
+  if (mode && !(mode === 'save' || mode === 'submit')) {
+    const GenericButton = getComponentFromMode(mode);
+    return (
+      <div className='container pl-0'>
+        <div className={css['button-group']}>
+          <div className={css['button-row']}>
+            <GenericButton {...props}>{props.text}</GenericButton>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  const saveFormData = () => {
+    dispatch(FormDataActions.submit({ componentId: 'saveBtn' }));
+  };
+
+  const submitForm = ({ componentId }: { componentId: string }) => {
+    const { org, app, instanceId } = window as Window as IAltinnWindow;
+    dispatch(
+      FormDataActions.submit({
```

Comment:
I'm pretty sure sure this won't actually work in navigating from the confirm page to completion (receipt). In the static confirm view, we had a custom submit button, which called `process.next`. But here, in order to display the same "submit" button, the layout must use this button, which submits the _form data_, which is not actually connected to the confirm task. The user will typically not have access to write to the data task from the confirm task, so this will cause a 403 response from the server. 
A quick fix might be to add a check in `submitFormDataSaga` and only PUT form data if the current task is a data task. Thoughts @lorang92 @TheTechArch ?

Reply:
Ah, I see that we got a `GoToTaskButton` which only calls `ProcessActions.complete`. I'm guessing this can be used in sted of the submit-button when we are in receipt going to completion?

Reply:
Can you confirm that this works @tjololo? I'm imagining we tried the whole flow from Task_1 to the end event?

Reply:
Yep that might work. But the styling won't be the same as the static confirm page where we have the green button.

Reply:
Good point. And I agree we should probably just fix so that submit-button works in confirm :)

Reply:
I have tested with the latest commit from this branch. There are some errors presented when I run `yarn start`
Other than that I have tested the whole process in an application with custom confirm page (and buttons) and an app with the default confirm page. Both seemed to work fine locally.
The "confirm" buttons on the custom confirm page is not styled as a submit button like the one on the static page, but that might my fault.
Error presented from yarn start:
```
ERROR in ./__mocks__/validationStateMock.ts:[32m[1m2:10[22m[39m

TS2459: Module '"../src/utils/textResource"' declares 'getTextResourceByKey' locally, but it is not exported.
    1 | import type { IValidations } from '../src/types';
  > 2 | import { getTextResourceByKey } from '../src/utils/textResource';
      |          ^^^^^^^^^^^^^^^^^^^^
    3 |
    4 | export function getMockValidationState(withFixed = false): IValidations {
    5 |   const fixed = withFixed ? [] : undefined;
 ```
 
 Screenshot of custom confirm page:
 
![image](https://user-images.githubusercontent.com/1145298/192786908-3dcf63a2-3672-46ce-8833-25dd689cd9e6.png)

Reply:
from the formlayout of the confirm page:
```json
{
    "id": "submit",
    "type": "Button",
    "mode": "go-to-task",
    "taskId": "EndEvent_1",
    "textResourceBindings": {
      "title": "button.confirm"
    },
    "grid": {
       "xs": 2
     }
 }
```
Errors are gone after the latest changes 🥳

Reply:
The go-to-task mode is intentionality  secondary by the way.
I thought the button was supposed to be in addition to other buttons on the page. And then the submit and confirm buttons where supposed to work as before.

Reply:
Yes, that was the idea @mjulstein. But the submit button from the custom confirm page didn't actually work as it saved the form data, as described in my original comment. I pushed a fix for this, so the submit button should be working on the custom confirm page now.